R/object-from-call.R

```diff
@@ -25,6 +25,14 @@ object_from_call <- function(call, env, block) {
 
 find_data <- function(name, env) {
   ns <- env_namespace(env)
+
+  if (identical(name, "_PACKAGE")) {
+    return(structure(
+      list(pkg_name = packageName(env),
+           base_path = getNamespaceInfo(ns, "path")),
```

Comment:
In `object_defaults` you might be able to do some manipulation of `x$srcref$filename`, but I don't know if that contains the full path